packages/plugin/src/PluginTypes.ts

```diff
@@ -0,0 +1,99 @@
+import type { TablePluginComponent } from './TablePlugin';
+
+export const PluginType = Object.freeze({
+  AUTH_PLUGIN: 'AuthPlugin',
+  DASHBOARD_PLUGIN: 'DashboardPlugin',
+  TABLE_PLUGIN: 'TablePlugin',
+});
+
+/**
+ * @deprecated Use DashboardPlugin instead
+ */
+export type LegacyDashboardPlugin = { DashboardPlugin: React.ComponentType };
+
+/**
+ * @deprecated Use AuthPlugin instead
+ */
+export type LegacyAuthPlugin = {
```

Comment:
This isn't quite right - right now filtering isn't picking up the keycloak auth plugin.
I think it should be something like:
```
export type LegacyAuthPlugin = {
  AuthPlugin: {
    Component: ...,
    isAvailable: ...
  }
};
```

Reply:
It turns out that the auth keycloak plugin is different from the core auth plugins. The core auth plugins match the type I have. Keycloak auth plugin matches the type you proposed